SocketIOClientSwift/SocketEngine.swift

```diff
@@ -33,6 +33,8 @@ public final class SocketEngine: NSObject, SocketEngineSpec, WebSocketDelegate {
     public private(set) var ws: WebSocket?
 
     public weak var client: SocketEngineClient?
+    
+    private weak var sessionDelegate: NSURLSessionDelegate?
```

Comment:
Why is this weak? the delegate won't have a strong reference to the engine so there's no risk of retain cycle.